src/TorchSharp/Utils/ShuffleGenerator.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+
+namespace TorchSharp.Utils
+{
+    public class ShuffleGenerator
```

Comment:
Should this really be public? I'm assuming that it's not really intended to be used outside of TorchSharp, but I could be wrong.